services/transactionpool/forward_transactions.go

```diff
@@ -0,0 +1,151 @@
+package transactionpool
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/crypto/signature"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/synchronization"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/orbs-network/orbs-spec/types/go/services/handlers"
+	"github.com/pkg/errors"
+	"sync"
+	"time"
+)
+
+type TransactionForwarderConfig interface {
+	NodePublicKey() primitives.Ed25519PublicKey
+	NodePrivateKey() primitives.Ed25519PrivateKey
+	TransactionPoolPropagationBatchSize() uint16
+	TransactionPoolPropagationBatchingTimeout() time.Duration
+}
+
+func (s *service) RegisterTransactionResultsHandler(handler handlers.TransactionResultsHandler) {
```

Comment:
why did you move the registration of txresults to the forward tx file ? it doesn't really belong here

Reply:
the thinking was that this file deals with transaction propagation, but I'm not sure about this anymore

Reply:
is transaction result same as forwarding ?

Reply:
what do you mean by that?